src/io/der/octet_aligned.rs

```diff
@@ -1,7 +1,161 @@
 use crate::io::buf::OctetBuffer;
 use crate::io::der::Error;
 use crate::io::der::{DistinguishedRead, DistinguishedWrite};
+use crate::io::per::unaligned::BitRead;
 
-impl DistinguishedRead for OctetBuffer {}
+impl BitRead for OctetBuffer {
```

Comment:
This was another mistake by me because I was trying to mimic your code.
I needed a non-perfect implementation to start correction with you.
I'll change it now